src/backend/distributed/utils/multi_partitioning_utils.c

```diff
@@ -130,6 +139,124 @@ worker_fix_pre_citus10_partitioned_table_constraint_names(PG_FUNCTION_ARGS)
 }
 
 
+/*
+ * fix_partition_shard_index_names fixes the index names of shards of partitions of
```

Comment:
I think what we do in `fix_partition_shard_index_names` is pretty complex due to several loops, though I really cannot find a simpler way anyway. So, I'd love to have a pseudo code here explains what is going on, like:

```py
foreach parentShard in shardListOfParentTable:
    foreach parentShardIndex on parentShard:
        foreach inheritedIndex on parentShardIndex:
            foreach partitionShard of parentShard:
                  foreach inheritedIndex on parentShardIndex:
                        if inheritedIndex does not have proper name:
                              Rename(inheritedIndex, newName)
 ```

And, maybe with that, we can understand the algorithm a little more, and see if we can avoid any of the loops. For example, we seem to have `     foreach inheritedIndex on parentShardIndex:` twice in the algorithm. Can we pass parameters to avoid the second one?

Reply:
(The above pseudo-code might not be very accurate, please ensure to write the accurate algorithm)

Reply:
The accurate version is the following, I couldn't figure out a way to avoid the loops:
```
foreach parentShard in shardListOfParentTableId:
    foreach parentIndex on parent:
	generate qualifiedParentShardIndexName -> parentShardIndex
        foreach inheritedPartitionIndex on parentIndex:
	    get table relation of inheritedPartitionIndex -> partitionId
            foreach partitionShard in shardListOfPartitionid:
		generate qualifiedPartitionShardName -> partitionShard
		generate newPartitionShardIndexName
		(the following happens in the worker node)
		foreach inheritedPartitionShardIndex on parentShardIndex:
		    if table relation of inheritedPartitionShardIndex is partitionShard:
		        if inheritedPartitionShardIndex does not have proper name:
			   Rename(inheritedPartitionShardIndex, newPartitionShardIndexName)
		        break
```